### PR TITLE
work on using URL for payload to acquire data

### PR DESCRIPTION
**Pull request by eddiebroad**
refactoring and refinement for making the "Import Selected Method Configuration" modal.
Now uses an additional react component.
Now uses it to make an AJAX call with the selected MC's URL to fetch its payload.
The payload contains detailed data necessary for the import.
A subsequent PR/commit will have the import call.

### CODE REVIEW

**Comment by eddiebroad** [comment]
I had an in-person meeting with Chet this morning.  I had an in-person meeting with David Mohs this morning.  Based on those meetings my understanding of the requirements has changed.  This results in my reinterpretation of the resulting/underlying GUI and code requirements.  Based on those meeting and my own judgement and my understanding of the requirements, I'm going to modify the code in this PR.  Until those changes are made, committed and pushed, this code is not ready for review nor merging.  I expect the changes to be ready by mid/early afternoon and maybe even before.

**Comment by eddiebroad** [comment]
I will re-comment when the PR is ready for review after the changes are committed and pushed.

**Comment by eddiebroad** [comment]
This PR is now ready for some review!

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs [hunk 1]
```diff
@@ -15,26 +15,19 @@
...
-                                            ["methodStoreMethod" "methodVersion"])
-                   "destination" destination-map}]
-    param-map))
-
+  [selected-conf-name
+   selected-conf-ns
+   selected-conf-snapId
+   dest-name
+   dest-namespace]
+  {
```
Comment (dmohs) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs]: join with next line

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs [hunk 2]
```diff
@@ -15,26 +15,19 @@
...
+   "configurationNamespace" selected-conf-ns
+   "configurationName" selected-conf-name
+   "configurationSnapshotId" selected-conf-snapId
+   "destinationNamespace" dest-namespace
+   "destinationName" dest-name
```
Comment (dmohs) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs]: join with next two lines

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs [hunk 3]
```diff
@@ -49,71 +42,86 @@
...
+
+
+(react/defc ReactImportSelectedMC
+  {:render (fn [{:keys [state props refs]}]
+             (let [selected-conf-name (get  props :initName)
+                   selected-conf-namespace (get props :initNamespace)
+                   selected-conf-snapshotId (get props :initSnapshotId)
+                   workspace (:workspace props)
+                   dest-workspace-name (get workspace "name")
+                   dest-workspace-namespace (get workspace "namespace")
```
Comment (dmohs) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs]: join

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs [hunk 4]
```diff
@@ -49,71 +42,86 @@
...
-                                         (js/alert
-                                           (str "Error in Method Configuration Import : "
-                                             returned-err-msg)))))})))}]
-                 [comps/Button
-                  {:title-text "cancel import"
-                   :icon :x
-                   :onClick (fn []
-                              (swap! state assoc :show-import-mc-modal? false))}]]])
+                 [ReactImportSelectedMC {
+                                     :initSelectedMC (get @state :selected-conf)
```
Comment (dmohs) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs]: hyphens to separate words

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs [hunk 5]
```diff
@@ -49,71 +42,86 @@
...
+                                     :initName (get-in @state [:selected-conf "name"])
+                                     :initNamespace (get-in @state [:selected-conf "namespace"])
+                                     :initSnapshotId (get-in @state [:selected-conf "snapshotId"])
+                                     :parental-state state
+                                     :workspace (get props :workspace)
```
Comment (dmohs) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs]: join

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs [hunk 6]
```diff
@@ -49,71 +42,86 @@
...
-                                             {:message (.-statusText xhr)}]
-                                         (js/alert
-                                           (str "Error in Method Configuration Import : "
-                                             returned-err-msg)))))})))}]
-                 [comps/Button
-                  {:title-text "cancel import"
-                   :icon :x
-                   :onClick (fn []
-                              (swap! state assoc :show-import-mc-modal? false))}]]])
+                 [ReactImportSelectedMC {
```
Comment (dmohs) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs]: join with next line or move opening brace to next line

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs [hunk 7]
```diff
@@ -49,71 +42,86 @@
...
+                 "Import a Method Configuration"
+                 [:hr]
+                 (get-in @state [:selected-conf "name"])[:br]
+                 (get-in @state [:selected-conf "namespace"]) [:br]
+                 (get-in @state [:selected-conf "snapshotId"])
+                 [:div {:style {:position "absolute" :right 2 :top 2}}
+                  [:div {:style {:backgroundColor (:button-blue style/colors) :color "#fff"
+                                 :padding "0.5em" :cursor "pointer"}
+                         :onClick (fn [] (swap! state assoc :show-import-mc-modal? false))}
+                   (icons/font-icon {:style {:fontSize "60%"}} :x)]]
```
Comment (dmohs) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs]: join

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs (hunk 4 above)
Comment (eddiebroad) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs]: you mean add hyphen between the letters?

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs [hunk 8]
```diff
@@ -49,71 +42,86 @@
     (range (rand-int 50))))
 
 
+
+
+
+(react/defc ReactImportSelectedMC
```
Comment (dmohs) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs]: This is an object, so should be named as such. I suggest `ModalImportOptionsAndButton` for now so it asks for refactoring.

**Comment by dmohs** [comment]
Coming along. Fix style problems and I'll take another look.

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs [hunk 9]
```diff
@@ -49,71 +39,79 @@
...
 
 
+(react/defc ModalImportOptionsAndButton
+  {:render (fn [{:keys [state props refs]}]
+             (let [selected-conf-name (get  props :init-Name)
+                   selected-conf-namespace (get props :init-Namespace)
+                   selected-conf-snapshotId (get props :init-SnapshotId)
+                   workspace (:workspace props)
+                   dest-workspace-name (get workspace "name")
+                   dest-workspace-namespace (get workspace "namespace")
```
Comment (dmohs) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs]: join with next line

**Comment by dmohs** [comment]
:+1: Nice! Fix last style problem then go ahead and merge.

**Pull request merged**